### PR TITLE
[Relay] Make check stricter by using Feature. Fixed multiple bugs.

### PR DESCRIPTION
**Pull request by MarisaKirisame**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.
@jroesch @junrushao1994 @zhiics please help review.
Also, do anyone know why test_pass_merge_composite suddenly failed? I hadnt touched any of the files, and it shouldnt call CPS/PE/AD.

### CODE REVIEW

**Comment by junrushao1994** [comment]
Please ping me when this PR is ready for review :-)

**Comment by MarisaKirisame** [comment]
@junrushao1994 it is. I am trying to fix the ci though.

File: include/tvm/relay/feature.h [hunk 1]
```diff
@@ -124,6 +125,13 @@ class FeatureSet {
    */
   bool is_subset_of(const FeatureSet& rhs) const { return ((*this) - rhs).bs_.none(); }
 
+  /*!
+   * \brief Pretty Print the FeatureSet.
+   *
+   * \return a string representation.
+   */
+  std::string Print() const;
```
Comment (junrushao1994) [inline include/tvm/relay/feature.h:133]: Maybe `ToString` sounds better?

File: tests/python/relay/test_pass_merge_composite.py [hunk 2]
```diff
@@ -999,14 +1000,4 @@ def _check_type_false(extract):
...
-    test_optional_pattern()
-    test_merge_order()
-    test_parallel_merge()
-    test_multiple_input_subgraphs()
-    test_reuse_call_merge()
-    test_tuple_get_item_merge()
-    test_pattern_with_check()
-    test_diamond_not_merge()
-    test_type_check()
+    #pytest.main([__file__])
```
Comment (junrushao1994) [inline tests/python/relay/test_pass_merge_composite.py:1003]: recover this?

**Comment by MarisaKirisame** [comment]
@wweic @vinx13 can you guys review?

**Pull request merged**